app/grandchallenge/algorithms/models.py

```diff
@@ -355,48 +334,37 @@ def _copy_output_files(self, *, container, base_dir: Path):
             logger.warning("Output directory is empty")
             return
 
-        # TODO: This thing should not interact with the database
-        result = Result.objects.create(job_id=self._job_id)
+        with TemporaryDirectory() as tmpdir:
+            input_files = set()
 
-        # Create the upload session but do not save it until we have the
-        # files
-        upload_session = RawImageUploadSession(
-            algorithm_result=result, creator=result.job.creator
-        )
-
-        images = []
+            for file in output_files:
+                tmpfile = Path(safe_join(tmpdir, file.relative_to(base_dir)))
+                tmpfile.parent.mkdir(parents=True, exist_ok=True)
 
-        for file in output_files:
-            new_uuid = uuid.uuid4()
+                with open(tmpfile, "wb") as outfile:
+                    infile = get_file(container=container, src=file)
+                    buffer = True
+                    while buffer:
+                        buffer = infile.read(1024)
+                        outfile.write(buffer)
 
-            django_file = File(get_file(container=container, src=file))
+                input_files.add(tmpfile)
 
-            staged_file = StagedFile(
-                user_pk_str="staging_conversion_user_pk",
-                client_id=self._job_id,
-                client_filename=file.name,
-                file_id=new_uuid,
-                timeout=timezone.now() + timedelta(hours=24),
-                start_byte=0,
-                end_byte=django_file.size - 1,
-                total_size=django_file.size,
+            importer_result = import_images(
+                files=input_files,
+                builders=[image_builder_mhd, image_builder_tiff],
```

Comment:
Why just these two image builders?

Reply:
For the algorithm image output we only want to support mha/mhd and tiff files.